src/satosa/base.py

```diff
@@ -222,7 +222,10 @@ def _load_state(self, context):
         try:
             state = cookie_to_state(context.cookie, self.config["COOKIE_STATE_NAME"],
                                     self.config["STATE_ENCRYPTION_KEY"])
-        except SATOSAStateError:
+        except Exception as error:
```

Comment:
`Exception` seems too generic here, maybe revert to 

```python
except (SATOSAStateError, ValueError) as error:
  ...
```

Reply:
That would cause a hard failure for the client (HTTP 500 without further explanation) to the user. I thought there might be cases in the context of an AuthnRequest where a state reset might heal the problem. Combined with proper exception monitoring issues could be ironed out this way. But I have no strong feelings about this.

Reply:
The error occurs within the `state.py` module. That module is responsible to handle the failure and expose it if needed as a generic `SATOSAStateError`. The error is logged and the state is reset.